packages/nextjs/src/server/data/getAuthDataFromRequest.ts

```diff
@@ -104,7 +103,12 @@ export const getAuthDataFromRequestAsync = async (
   };
 
   const hasMachineToken = bearerToken && isMachineTokenByPrefix(bearerToken);
-  if (hasMachineToken) {
+
+  const acceptsOnlySessionToken =
+    acceptsToken === TokenType.SessionToken ||
+    (Array.isArray(acceptsToken) && acceptsToken.length === 1 && acceptsToken[0] === TokenType.SessionToken);
```

Comment:
handles case where some passes in an array of accepted tokens, but only provided a session token:

```ts
auth({ acceptsToken: ['session_token'] })
```